### PR TITLE
Fix pos-only params in `os` module

### PR DESCRIPTION
**Pull request by sobolevn**
Problem:

```
/home/runner/work/typeshed/typeshed/stdlib/os/__init__.pyi
  /home/runner/work/typeshed/typeshed/stdlib/os/__init__.pyi:872:33 - error: Position-only parameter not allowed after parameter that is not position-only (reportGeneralTypeIssues)
  /home/runner/work/typeshed/typeshed/stdlib/os/__init__.pyi:873:34 - error: Position-only parameter not allowed after parameter that is not position-only (reportGeneralTypeIssues)
```

It is the same notation for `__arg1: X, *args: X`
Link: https://mypy-play.net/?mypy=latest&python=3.12&gist=b35729b668bf88a0d98a78a8d37c5be1

### CODE REVIEW

File: stdlib/os/__init__.pyi [hunk 1]
```diff
@@ -869,8 +869,8 @@ if sys.platform != "win32":
 def abort() -> NoReturn: ...
 
 # These are defined as execl(file, *args) but the first *arg is mandatory.
-def execl(file: StrOrBytesPath, __arg0: StrOrBytesPath, *args: StrOrBytesPath) -> NoReturn: ...
-def execlp(file: StrOrBytesPath, __arg0: StrOrBytesPath, *args: StrOrBytesPath) -> NoReturn: ...
+def execl(file: StrOrBytesPath, *args: Unpack[tuple[StrOrBytesPath, Unpack[tuple[StrOrBytesPath, ...]]]]) -> NoReturn: ...
+def execlp(file: StrOrBytesPath, *args: Unpack[tuple[StrOrBytesPath, Unpack[tuple[StrOrBytesPath, ...]]]]) -> NoReturn: ...
```
Comment (AlexWaygood) [inline stdlib/os/__init__.pyi:873]: Using PEP646 feels unnecessarily complicated here (and e.g. pytype still doesn't support PEP646 -- it just falls back to `Any` when it sees `Unpack`, I believe). Can't we just do this?

```suggestion
def execl(file: StrOrBytesPath, arg0: StrOrBytesPath, /, *args: StrOrBytesPath) -> NoReturn: ...
def execlp(file: StrOrBytesPath, arg0: StrOrBytesPath, /, *args: StrOrBytesPath) -> NoReturn: ...
```

Comment (sobolevn) [reply stdlib/os/__init__.pyi:873]: Yes, looks easier!

**Comment by AlexWaygood** [comment]
Oh interesting. Stubtest is correct; you really can pass `file` as a keyword argument:

```pycon
>>> import os
>>> os.execl(file='foo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<frozen os>", line 548, in execl
ValueError: execv() arg 2 must not be empty
```

**Comment by sobolevn** [comment]
Hm, this way we have a stubtest problem:

```
error: os.execl is inconsistent, stub argument "file" should be positional or keyword (remove leading double underscore)
Stub: in file stdlib/os/__init__.pyi:872
def (Union[builtins.str, builtins.bytes, os.PathLike[builtins.str], os.PathLike[builtins.bytes]], Union[builtins.str, builtins.bytes, os.PathLike[builtins.str], os.PathLike[builtins.bytes]], *args: Union[builtins.str, builtins.bytes, os.PathLike[builtins.str], os.PathLike[builtins.bytes]]) -> Never
Runtime: in file /opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/os.py:538
def (file, *args)

error: os.execlp is inconsistent, stub argument "file" should be positional or keyword (remove leading double underscore)
Stub: in file stdlib/os/__init__.pyi:873
def (Union[builtins.str, builtins.bytes, os.PathLike[builtins.str], os.PathLike[builtins.bytes]], Union[builtins.str, builtins.bytes, os.PathLike[builtins.str], os.PathLike[builtins.bytes]], *args: Union[builtins.str, builtins.bytes, os.PathLike[builtins.str], os.PathLike[builtins.bytes]]) -> Never
Runtime: in file /opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/os.py:553
def (file, *args)
```

**Comment by AlexWaygood** [comment]
I guess your PEP646 way is the best way, in that case :)

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**